src/output.c

```diff
@@ -103,6 +104,20 @@ static void mkdir_exist_ok(const char *p)
     exit(1);
 }
 
+static void write_gitinore(const char *p)
+{
+    char *filename = malloc_sprintf("%s/.gitinore", p);
+    if (access(filename, F_OK)) {
+        free(filename);
+        return;
+    }
+    FILE *gitinore = fopen(filename, "w");
+    fprintf(gitinore, "*");
+
+    fclose(gitinore);
```

Comment:
On Windows, opening the file for writing fails if another process already has it opened. This can happen in various surprising ways:
- another instance of Jou compiler is currently writing `*` into it
- user's editor is running git in the background
- some other program is reading the `.gitignore` (many programs do).

If you don't check for NULL, trying to use `gitinore` will crash the compiler when opening fails.

```suggestion
    if (gitinore != NULL) {
        fprintf(gitinore, "*");
        fclose(gitinore);
    }
```

You might think that the `access()` check solves this, but it doesn't quite. Another program could create the file between `access()` and `fopen()`.